### PR TITLE
Update to Bevy 0.13 

### PR DESCRIPTION
**Pull request by dgsantana**
Since we are working on a simple game that is using replicon, and we would like to switch to 0.13, to catch any early bugs and take advantage of the new features, I made this PR.

Changes:
- rennet PR to bring bevy_rennet to bevy 0.13 (https://github.com/lucaspoffo/renet/pull/147)
- deprecated `MapNetworkEntities `-> `MapEntites` and `Mapper`->`EntityMapper`
- condition systems to new bevy pattern.

Missing review of entity id serialization, couldn't find where it was.

### CODE REVIEW

**Review by Shatur — CHANGES_REQUESTED** [review]
Thank you a lot!

Could you please apply formatting, fix doctests and silence deprecation warnings for existing `Mapper` impls to make CI happy?
Also I would appreciate if you update the changelog :)

But it's all optional, I can do all of the above myself later.

File: Cargo.toml
Comment (dgsantana) [reply Cargo.toml:59]: Of course no problem at all.

**Review by dgsantana — COMMENTED** [review]
Fixed version and missing extra LF

File: tests/other.rs
Comment (dgsantana) [inline tests/other.rs:94]: Here I changed from 33 to 38. Don't know if this is from the id's changed on the bevy size, or something on rennet.

File: CHANGELOG.md
Comment (Shatur) [inline CHANGELOG.md:16]: Changes need to be placed after `[Unreleased]`

**Review by Shatur — CHANGES_REQUESTED** [review]
Thanks!

Since we changed our generics to accept `MapEntities` instead of `MapNetworkEntities`, it will stop to compile anyway. Maybe remove this trait instead of deprecation?

File: CHANGELOG.md
Comment (dgsantana) [reply CHANGELOG.md:16]: Oh, sorry my bad.

**Comment by dgsantana** [comment]
There is a test on the other.rs that as 5 more bytes on the stats. I silence the assert, since I don't see any regressions, and maybe something from the rennet update.

> Thanks!
> 
> Since we changed our generics to accept `MapEntities` instead of `MapNetworkEntities`, it will stop to compile anyway. Maybe remove this trait instead of deprecation?

I agree, I just deprecated mostly because of the comment on #185.

**Comment by Shatur** [comment]
> I agree, I just deprecated mostly because of the comment on https://github.com/projectharmonia/bevy_replicon/issues/185.

Let's ask @UkoeHB about it. Do you think it worth to remove the old mapper trait since all methods now accept built-in?

**Comment by dgsantana** [comment]
Of course. Also bevy_renet should be ready to be merged. Bevy_egui is now published.

**Comment by Shatur** [comment]
> Also bevy_renet should be ready to be merged. Bevy_egui is now published.

Great! But `bevy_renet` is maintained by a different person, so we need to wait until he merges the PR.

**Comment by Shatur** [comment]
@UkoeHB could you take a look into entity serialization?

**Comment by dgsantana** [comment]
There is something broken on component serialization, the client receives always 0 bytes after the entity id. I'm debugging on a different branch.

**Comment by Shatur** [comment]
Make sure to apply changes from `@UkoeHB`. I will fix the CI and merge his branch today after work.

**Comment by dgsantana** [comment]
> Make sure to apply changes from `@UkoeHB`. I will fix the CI and merge his branch today after work.

I did, but there is something broken probably on the collect_changes on the server... I have to leave for a few hours, but will look into this later on.

**Comment by Shatur** [comment]
Pushed some minor changes to make CI happy.

> I did, but there is something broken probably on the collect_changes on the server.

Are you sure? Because all tests pass and all examples works.

**Review by Shatur — APPROVED** [review]

**Comment by dgsantana** [comment]
I will double check again. But the data buffer after the entity is empty on
the client side. I should be back in a few hours.

A terça, 20/02/2024, 16:46, codecov[bot] ***@***.***>
escreveu:

**Comment by Shatur** [comment]
Great!
I will merge when @dgsantana confirms that everything is fine and Renet drafts a new release.

**Comment by dgsantana** [comment]
> Great! I will merge when @dgsantana confirms that everything is fine and Renet drafts a new release.

I'm pulling all changes from the current PR into my instrumented branch just to double check everything.

**Comment by dgsantana** [comment]
It's very weird...
image
```
2024-02-20T18:32:40.837400Z DEBUG bevy_replicon::client: applying Removal components for 0 entities
2024-02-20T18:32:40.837642Z DEBUG bevy_replicon::client: applying Insert components for 2 entities
2024-02-20T18:32:40.837811Z DEBUG bevy_replicon::client: reading components for 13v2 with 0 bytes
2024-02-20T18:32:40.838129Z DEBUG bevy_replicon::client: reading components for 14v2 with 0 bytes
```

**Comment by Shatur** [comment]
Could you provide a minimal example to reproduce?
Preferably in a form of test :) But any other form is also fine.

**Comment by dgsantana** [comment]
> Could you provide a minimal example to reproduce? Preferably in a form of test :) But any other form is also fine.

I will try to create a minimal sample or test for this.

**Comment by dgsantana** [comment]
Ok here goes a "minimal" example. It uses a menu system to handle the connection, instead of the command line. And it as the issue. Feel free to use as a extra example.
[minimal_with_menu.zip](https://github.com/projectharmonia/bevy_replicon/files/14350054/minimal_with_menu.zip)

**Comment by dgsantana** [comment]
> @dgsantana it looks like you are not registering any components for replication. [Docs](https://docs.rs/bevy_replicon/0.22.0/bevy_replicon/#marking-for-replication)

Ok, I made it too minimal. I removed serde by mistake.
Player, PlayerStone in the original code as Serialize and Deserialize.
image

**Comment by dgsantana** [comment]
Oook, manually registering with `replicate::<...>()` works.

**Comment by dgsantana** [comment]
Ok it was my bad, I indeed forgot to register the components for replication, sorry for the false alarm.

**Pull request merged**
